### PR TITLE
Update nextjs app router guide to use ai/rsc

### PR DESCRIPTION
**Pull request by jeremyphilemon**

### CODE REVIEW

File: docs/pages/docs/guides/frameworks/nextjs-app.mdx [hunk 1]
```diff
@@ -8,169 +8,95 @@ import { Callout } from 'nextra-theme-docs';
...
-
-Below is a minimal route handler for using the OpenAI Chat API with the `openai` API client and the Vercel AI SDK. Consult our [guides](/docs/guides) for examples of using other providers.
+```typescript filename="app/page.tsx" showLineNumbers
+'use client';
 
-```typescript filename="app/api/chat/route.ts" showLineNumbers
-import OpenAI from 'openai';
-import { OpenAIStream, StreamingTextResponse } from 'ai';
+import { useState } from 'react';
+import { generate } from './actions';
```
Comment (jaredpalmer) [inline docs/pages/docs/guides/frameworks/nextjs-app.mdx:19]: should we call this generate? given `generateText` coming?

Comment (jaredpalmer) [inline docs/pages/docs/guides/frameworks/nextjs-app.mdx:19]: should we call this generate? given `generateText` coming?

File: docs/pages/docs/guides/frameworks/nextjs-app.mdx [hunk 2]
```diff
@@ -8,169 +8,95 @@ import { Callout } from 'nextra-theme-docs';
...
-  const response = await openai.chat.completions.create({
-    model: 'gpt-4',
-    stream: true,
+export async function generate(input: string) {
+  'use server';
+
+  // Creates a wrapped, changable value that can be streamed to the client.
+  const stream = createStreamableValue('');
+
+  render({
```
Comment (jaredpalmer) [inline docs/pages/docs/guides/frameworks/nextjs-app.mdx:73]: why not have `const ui = render()` and then return `ui` from the stream?

File: docs/pages/docs/guides/frameworks/nextjs-app.mdx [hunk 3]
```diff
@@ -8,169 +8,95 @@ import { Callout } from 'nextra-theme-docs';
...
-      <form onSubmit={handleSubmit}>
-        <label>
-          Say something...
-          <input value={input} onChange={handleInputChange} />
-        </label>
-        <button type="submit">Send</button>
-      </form>
-    </div>
-  );
+  return {
```
Comment (jaredpalmer) [inline docs/pages/docs/guides/frameworks/nextjs-app.mdx:97]: Missing other APIs and or links to them.

File: docs/pages/docs/guides/frameworks/nextjs-app.mdx [hunk 4]
```diff
@@ -8,169 +8,95 @@ import { Callout } from 'nextra-theme-docs';
...
+        Ask
+      </button>
+
+      <div>{generation}</div>
+    </div>
+  );
 }
 ```
 
 Note that using the Edge Runtime is optional but highly recommended due to [longer streaming timeouts on Vercel](https://vercel.com/docs/concepts/functions/edge-functions/limitations#maximum-initial-response-time), no cold starts, and lower latency.
```
Comment (jaredpalmer) [inline docs/pages/docs/guides/frameworks/nextjs-app.mdx:53]: Remove this

**Review by jaredpalmer — CHANGES_REQUESTED** [review]

**Review by c45 — APPROVED** [review]

**Comment by lgrammel** [comment]
@jeremyphilemon can this be closed?

**Pull request closed without merging**